src/shogun/converter/ica/SOBI.cpp

```diff
@@ -66,8 +66,11 @@ void CSOBI::fit_dense(CDenseFeatures<float64_t>* features)
 	MatrixXd M0 = cor(EX,int(m_tau[0]));
 	EigenSolver<MatrixXd> eig;
 	eig.compute(M0);
-	MatrixXd SPH = (eig.pseudoEigenvectors() * eig.pseudoEigenvalueMatrix().cwiseSqrt() * eig.pseudoEigenvectors ().transpose()).inverse();
-	MatrixXd spx = SPH*EX;
+	MatrixXd EVMsqrt = eig.pseudoEigenvalueMatrix().cwiseSqrt();
```

Comment:
just asking, is this covered by any tests that would detect changing results?

Reply:
Should be covered [here](https://github.com/shogun-toolbox/shogun/blob/41888fe7c8dc3797063d674f452e13351f321338/tests/unit/converter/ica/SOBI_unittest.cc). That's how I found these bugs!

Reply:
Is there a buildbot for MKL backend btw?